pkg/controller/routes.go

```diff
@@ -111,6 +111,7 @@ func routes(router *router.Router, cfg *rest.Config, registryTransport http.Roun
 	router.Type(&corev1.Pod{}).Selector(managedSelector).HandlerFunc(jobsHandler.SaveJobOutput)
 	router.Type(&netv1.Ingress{}).Selector(managedSelector).Namespace(system.ImagesNamespace).HandlerFunc(gc.GCOrphans)
 	router.Type(&netv1.Ingress{}).Selector(managedSelector).Middleware(ingress.RequireLBs).Handler(ingress.NewDNSHandler())
+	router.Type(&corev1.Secret{}).Selector(managedSelector).HandlerFunc(secrets.HandleDNSSecret)
```

Comment:
instead of watching all secrets, maybe we should use the Namesapce() route function to limit it.

Reply:
same with the `NewDNSHandler`route. might as well limit that to the the system namesapce too?